src/app/organizations/state/register-organization.service.spec.ts

```diff
@@ -0,0 +1,73 @@
+
+describe('RegisterOrganizationService', () => {
+  let registerOrganizationService: RegisterOrganizationService;
+  const exampleFormState = { name: '', description: '', link: '', location: '', contactEmail: '' };
+  beforeEach(() => {
+    const organisationsServiceStub = jasmine.createSpyObj('OrganizationsService', ['createOrganisation']);
+    const matDialogStub = jasmine.createSpyObj('MatDialog', ['closeAll']);
+    TestBed.configureTestingModule({
+      providers: [RegisterOrganizationService,
+        { provide: OrganisationsService, useValue: organisationsServiceStub },
+        { provide: MatDialog, useValue: matDialogStub }
+      ],
+      imports: [HttpClientTestingModule, MatSnackBarModule, MatDialogModule, BrowserAnimationsModule, RouterTestingModule]
+    });
+
+    registerOrganizationService = TestBed.get(RegisterOrganizationService);
+    organisationsServiceSpy = TestBed.get(OrganisationsService);
+    matDialogSpy = TestBed.get(MatDialog);
+  });
+
+  it('should be created', () => {
+    expect(registerOrganizationService).toBeDefined();
+  });
+
+  it('should try to add an organization', () => {
+    organisationsServiceSpy.createOrganisation.and.returnValue(observableOf(null));
+    matDialogSpy.closeAll.and.returnValue(null);
+
+    registerOrganizationService.addOrganization(exampleFormState);
+
+    // Expected createOrganisation call to be called (and it will succeed)
+    expect(organisationsServiceSpy.createOrganisation.calls.count()).toBe(1, 'spy method was called once');
+    // Upon success, the dialog would close
+    expect(matDialogSpy.closeAll.calls.count()).toBe(1, 'spy method was not called once');
+  });
+
+  it('should handle error when adding an organization', () => {
+    organisationsServiceSpy.createOrganisation.and.returnValue(throwError('test 404 error'));
+
+    registerOrganizationService.addOrganization(exampleFormState);
+
+    // Expected createOrganisation call to be called (even though it will fail)
+    expect(organisationsServiceSpy.createOrganisation.calls.count()).toBe(1, 'spy method was called once');
+    // Upon error, the dialog would not close
+    expect(matDialogSpy.closeAll.calls.count()).toBe(0, 'spy method was called');
+  });
+
+  it('should have the correct URL regex', () => {
+    const regexp = new RegExp(registerOrganizationService.urlRegex);
+    expect(regexp.test('https://www.google.ca')).toBeTruthy();
+    expect(regexp.test('http://www.google.ca')).toBeTruthy();
+    expect(regexp.test('https://google.ca')).toBeTruthy();
+    expect(regexp.test('http://google.ca')).toBeTruthy();
+    expect(regexp.test('http://google.com')).toBeTruthy();
+    expect(regexp.test('http://google.ca/potato')).toBeTruthy();
+    expect(regexp.test('http://numb3r5.ca')).toBeTruthy();
+    expect(regexp.test('httpz://google.ca')).toBeFalsy();
+    expect(regexp.test('httpsz://google.ca')).toBeFalsy();
+    expect(regexp.test('https://google.tooLongTLD')).toBeFalsy();
```

Comment:
Probably too restrictive
Consider http://nic.yodobashi/
Or 
https://stackoverflow.com/questions/9238640/how-long-can-a-tld-possibly-be